src/ir/named.rs

```diff
@@ -359,33 +359,47 @@ impl<'ctx, 'gen> MonotoneFramework for UsedTemplateParameters<'ctx, 'gen>  {
     }
 
     fn constrain(&mut self, id: ItemId) -> bool {
-        let original_len = self.used[&id].len();
+        // Invariant: all hash map entries' values are `Some` upon entering and
+        // exiting this method.
+        debug_assert!(self.used.values().all(|v| v.is_some()));
```

Comment:
Ugh, you're right :(

I think we should probably define our own `testing_assert!(...)` macro that only runs under tests and replace **all** of our debug asserts with that.

Follow up?